requirements/doc-min-requirements.txt

```diff
@@ -1,6 +1,5 @@
-# sphinx >= 4.0.0 is incompatible with our custom CSS styles and renders the documents improperly.
-# See https://github.com/mlflow/mlflow/pull/4480
-sphinx==3.5.4
+# Minimum version that works with Python 3.10
+sphinx==4.2.0
```

Comment:
The API reference looks fine: https://output.circle-artifacts.com/output/job/725f1eeb-71e5-41ed-91b6-5a1aecaa36ed/artifacts/0/docs/build/latest/api_reference/index.html